### DISCUSSION TITLE
image-reflector-controller  scan failed: Bad credentials

### DISCUSSION THREAD

**Discussion by 00787718** [fluxcd/flux2#4016] [Q&A]
I am trying to create flux image repository for image automation with private image repository. I am able to call the repository api to get the build tags using credentials from postman . but not able to reconcile the image repository using docker-registry secret created with same credential. I am getting this below Warning  ReadOperationFailed  2s (x4 over 13s)  image-reflector-controller  scan failed: GET https://pppp.xxxx.com/artifactory/api/docker/null/v2/token?scope=repository%!!(MISSING)!(MISSING)A(MISSING)artifactory%!!(MISSING)!(MISSING)F(MISSING)api%!!(MISSING)!(MISSING)F(MISSING)docker%!!(MISSING)!(MISSING)F(MISSING)XXXXXXX%!!(MISSING)!(MISSING)F(MISSING)v2%!!(MISSING)!(MISSING)F(MISSING)get%!!(MISSING)!(MISSING)F(MISSING)XXXXXXXXXX%!!(MISSING)!(MISSING)F(MISSING)tags%!!(MISSING)!(MISSING)F(MISSING)list%!!(MISSING)!(MISSING)A(MISSING)pull&service=pppp.xxxx.com: : Bad credentials

**Comment by kingdonb** [comment]
This looks like an authentication failure. My first question is, how did you create the Docker Registry secret?

In the guide, it uses `kubectl create secret docker-registry` – it sounds like you used that. If you can test that registry with the imagePullSecret using a Kubernetes deployment or pod, are you able to verify the credentials are usable there?

If the docker-registry secret works as an image pull secret as described in https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/ then it should work with Flux. Please let us know if you can test that method, and resolve any issues there, then if Kubernetes will work with the secret, let us know if Flux won't also work with the secret.
